pom.xml

```diff
@@ -397,6 +398,20 @@
                 <version>${log4j.version}</version>
             </dependency>
             <dependency>
+              <!-- This is not slf4j's version because of performance concerns
+              http://www.slf4j.org/api/org/slf4j/bridge/SLF4JBridgeHandler.html
+              Please make sure to do performance tests before switching this to slf4j
+              -->
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-jul</artifactId>
+                <version>${log4j.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.slf4j</groupId>
+                <artifactId>jcl-over-slf4j</artifactId>
+                <version>${slf4j.version}</version>
```

Comment:
The soft-goal is to reduce the dependency on log implementation, which is why most logging in Druid is slf4j based, and we just happen to use the log4j implementation. It *should* be directly pluggable with Logback (assuming proper config of course). The slf4j team has raised no concerns about performance of jcl-over-slf4j that I have seen, but were very explicit about performance problems in the jul bridge.

The log4j-jul breaks this general goal. Notably there is no note either way from the log4j2 folks regarding log4j-jul performance, which is concerning. If anyone has an example of performance differences that would be very helpful.

 Basically I would be proposing the slf4j jul bridge if it weren't for the performance concern.